sdks/java/io/cassandra/src/test/java/org/apache/beam/sdk/io/cassandra/CassandraIOIT.java

```diff
@@ -0,0 +1,171 @@
+ *
+ * <pre>{@code
+ * mvn -e -Pio-it verify -pl sdks/java/io/cassandra -DintegrationTestPipelineOptions='[
+ * "--cassandraHost=1.2.3.4",
+ * "--cassandraPort=7001"]'
+ * }</pre>
+ */
+@RunWith(JUnit4.class)
+public class CassandraIOIT {
+
+  private static IOTestPipelineOptions options;
+
+  @Rule public TestPipeline pipeline = TestPipeline.create();
+
+  @BeforeClass
+  public static void setup() throws Exception {
+    PipelineOptionsFactory.register(IOTestPipelineOptions.class);
+    options = TestPipeline.testingPipelineOptions()
+        .as(IOTestPipelineOptions.class);
+  }
+
+  @Test
+  public void testRead() throws Exception {
+    PCollection<Scientist> output = pipeline.apply(CassandraIO.<Scientist>read()
+        .withHosts(Collections.singletonList(options.getCassandraHost()))
+        .withPort(options.getCassandraPort())
+        .withKeyspace(CassandraTestDataSet.KEYSPACE)
+        .withTable(CassandraTestDataSet.READ_TABLE_NAME)
+        .withEntity(Scientist.class)
+        .withCoder(SerializableCoder.of(Scientist.class)));
+
+    PAssert.thatSingleton(output.apply("Count scientist", Count.<Scientist>globally()))
+        .isEqualTo(1000L);
+
+    PCollection<KV<String, Integer>> mapped =
+        output.apply(
+            MapElements.via(
+                new SimpleFunction<Scientist, KV<String, Integer>>() {
+                  public KV<String, Integer> apply(Scientist scientist) {
+                    KV<String, Integer> kv = KV.of(scientist.name, scientist.id);
+                    return kv;
+                  }
+                }
+            )
+        );
+    PAssert.that(mapped.apply("Count occurences per scientist", Count.<String, Integer>perKey()))
+        .satisfies(
+            new SerializableFunction<Iterable<KV<String, Long>>, Void>() {
+              @Override
+              public Void apply(Iterable<KV<String, Long>> input) {
+                for (KV<String, Long> element : input) {
+                  assertEquals(element.getKey(), 1000 / 10, element.getValue().longValue());
+                }
+                return null;
+              }
+            }
+        );
+
+    pipeline.run().waitUntilFinish();
+  }
+
+  @Test
+  public void testWrite() throws Exception {
+    String tableName = CassandraTestDataSet.createWriteDataTable(options);
+
+    ArrayList<Scientist> data = new ArrayList<>();
+    for (int i = 0; i < 1000; i++) {
+      Scientist scientist = new Scientist();
+      scientist.id = i;
+      scientist.name = "Name " + i;
+      data.add(scientist);
+    }
+
+    pipeline
+        .apply(Create.of(data))
+        .apply(CassandraIO.<Scientist>write()
+            .withHosts(Collections.singletonList(options.getCassandraHost()))
+            .withPort(options.getCassandraPort())
+            .withKeyspace(CassandraTestDataSet.KEYSPACE)
+            .withEntity(Scientist.class));
+
+    pipeline.run().waitUntilFinish();
+
+    Cluster cluster = CassandraTestDataSet.getCluster(options);
```

Comment:
It looks like this section could be generalized into an OnSuccessMatcher, which is something we added to TestPipelineOptions a while ago. You can see an example in WordCountIT (using a generalized FileChecksumMatcher class). WDYT?